rootfs.go

```diff
@@ -14,23 +14,76 @@
 package main
 
 import "fmt"
+import "net/http"
 import "io"
 import "os"
+import "strings"
 
 func doRootfs(imageFile string) error {
+	var imageFd io.Reader
+	var imageSize int64
+	var err error
+
+	if strings.HasPrefix(imageFile, "http:") ||
+		strings.HasPrefix(imageFile, "https:") {
+		// Network based update.
+		resp, err := http.Get(imageFile)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		imageSize = resp.ContentLength
+		if imageSize < 0 {
+			return fmt.Errorf("Image size from '%s' is unknown. "+
+				"Will not continue with unknown image size.",
+				imageFile)
+		} else if imageSize < 4096 {
```

Comment:
4K is quite arbitrary, it's just a safeguard against unreasonably small updates. This is just a safety net, not a guaranteed check.

But yes, I can use a constant.